final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,123 @@
+    """
+    colorama.init()
+    arguments = get_arguments()
+    if arguments.version:
+        print(f'Program version - {VERSION}')
+        return
+    if arguments.verbose:
+        logging.basicConfig(level=logging.INFO)
+    else:
+        logging.basicConfig(filename='sample.log', filemode='w', level=logging.INFO)
+    logging.info('Program started')
+    if arguments.to_html:
+        if not '.html' in arguments.to_html or \
+                not os.access(os.path.dirname(arguments.to_html), os.W_OK) and '\\' in arguments.to_html:
+            logging.error('Inrorrect html filepath')
+            raise IncorrectFilePath('Inrorrect html filepath')
+    if arguments.to_fb2:
+        if not '.fb2' in arguments.to_fb2 or \
+                not os.access(os.path.dirname(arguments.to_fb2), os.W_OK) and '\\' in arguments.to_fb2:
+            logging.error('Inrorrect fb2 filepath')
+            raise IncorrectFilePath('Inrorrect fb2 filepath')
+    if arguments.limit:
+        if not re.match('\\d+', arguments.limit):
+            logging.error('Input value of --limit is incorrect')
+            raise ValueError('Input value of --limit is incorrect')
+        arguments.limit = int(arguments.limit)
+    if arguments.date:
+        if not re.match('\\d+', arguments.date) or len(arguments.date) != 8:
+            logging.error('Input value of --date is incorrect')
+            raise ValueError('Input value of --date is incorrect')
+        news_list = get_news_list_by_date(arguments.date, arguments.limit)
+        if arguments.to_html or arguments.to_fb2:
+            if arguments.to_html:
+                save_in_html(arguments.to_html, news_list)
+            if arguments.to_fb2:
+                save_in_fb2(arguments.to_fb2, news_list)
+        else:
+            if news_list:
+                if arguments.colorize:
+                    if arguments.json:
+                        print_news_JSON_colorize(news_list)
+                    else:
+                        print_news_colorize(news_list)
+                else:
+                    if arguments.json:
+                        print_news_JSON(news_list)
+                    else:
+                        print_news(news_list)
+            else:
+                print('No news by this date')
+        return
+    news_list = get_news_list(arguments.source, arguments.limit)
+    if arguments.to_html or arguments.to_fb2:
+        if arguments.to_html:
+            save_in_html(arguments.to_html, news_list)
+        if arguments.to_fb2:
+            save_in_fb2(arguments.to_fb2, news_list)
+    else:
+        if arguments.colorize:
+            if arguments.json:
+                print_news_JSON_colorize(news_list)
+            else:
+                print_news_colorize(news_list)
+        else:
+            if arguments.json:
+                print_news_JSON(news_list)
+            else:
+                print_news(news_list)
+    write_news_to_database(news_list)
```

Comment:
It is better to split this function a bit, for example you can move each argument check to a separate function: `check_html`, `check_fb2` and etc

Reply:
Don't understent how it can be splitted

Reply:
for example 
```if arguments.to_html:
       check_html_arg(arguments.to_html)
   if arguments....
```

Reply:
for example:
```
if arguments.to_html:
    check_html_arg(arguments.html)
if arguments.....
```